### ISSUE TITLE
Different result presented in os.arch between openjdk and openj9 on Mac

### ISSUE THREAD

**Issue by kiszk** [eclipse/openj9#5493]
I realized that `os.arch` is different between openjdk and openj9 on Mac. On Linux, they return the same `os.arch`.

```
$ /Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home/bin/java -XshowSettings:all
...
    os.arch = x86_64
...
$ /Library/Java/JavaVirtualMachines/adoptopenjdk-8-openj9.jdk/Contents/Home/bin/java -XshowSettings:all
...
    os.arch = amd64
...
```

This difference causes [an JNI link error](https://github.com/apache/spark/pull/24308#pullrequestreview-223611574) in Spark. This is because `zstd-jni` determines [dll path](https://github.com/luben/zstd-jni/blob/master/src/main/java/com/github/luben/zstd/util/Native.java#L44) by using `os.arch` as a part of the path name.

**Comment by keithc-ca** [comment]
This seems to be platform specific: on a Linux x86 system:
```
$ ./bin/java -version
openjdk version "1.8.0_202"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_202-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.202-b08, mixed mode)
$ ./bin/java -XshowSettings:properties |& grep os.arch
    os.arch = amd64
```
It's surprising to me that hotspot would use a different name on MacOSX for the same CPU architecture.

**Comment by kiszk** [comment]
Yes, this problem occurs only on Mac.

**Comment by DanHeidinga** [comment]
Property is set in https://github.com/eclipse/openj9/blob/f0820b5ca51197ee3e58925920cdfc9fe5dc5220/runtime/vm/vmprops.c#L778-L782

**Comment by DanHeidinga** [comment]
Which is set in OMR:
https://github.com/eclipse/omr/blob/5edfbe1a6e36d6b462283b2aaee16147cb570c52/port/unix/omrsysinfo.c#L524-L525

and we define `J9HAMMER` for OSX:
https://github.com/eclipse/omr/blob/5edfbe1a6e36d6b462283b2aaee16147cb570c52/omrmakefiles/rules.osx.mk#L27

**Comment by DanHeidinga** [comment]
To fix this we either need to 
1) update `j9sysinfo_get_CPU_architecture` / `omrsysinfo_get_CPU_architecture` to return `x86_64` and check all the callers are ready to deal with this string, or
2) change vmprops.c to map from `amd64` to `x86_64` on osx.

**Comment by kiszk** [comment]
Thank you for your investigation and comments.  
IMHO, 2. is preferable to me.

**Comment by pshipton** [comment]
@keithc-ca can you please open a PR to fix this for the 0.14.2 release.

**Comment by keithc-ca** [comment]
> can you please open a PR to fix this for the 0.14.2 release

Done, see #5771.
